streams/src/test/java/org/apache/kafka/test/InternalMockProcessorContext.java

```diff
@@ -303,22 +305,21 @@ public Headers headers() {
     }
 
     public StateRestoreListener getRestoreListener(final String storeName) {
-        final BatchingStateRestoreCallback restoreCallback = getBatchingRestoreCallback(restoreFuncs.get(storeName));
-        return getStateRestoreListener(restoreCallback);
+        return getStateRestoreListener(restoreFuncs.get(storeName));
```

Comment:
We actually never should have been adapting it to a `BatchingStateRestoreCallback` before casting it to a `StateRestoreListener`.

The only reason it worked is that all our implementaitons happen to be `BatchingStateRestoreCallback` already, and the adaptation was therefore a no-op.